### PR TITLE
CLOUD-16 Endpoint for uploading wars

### PR DESCRIPTION
**Pull request by Cousjava**

### CODE REVIEW

File: deployer/deployment-controller/pom.xml [hunk 1]
```diff
@@ -109,90 +110,49 @@
...
+                                    <destFileName>payara-micro-${payara.version}.jar</destFileName>
                                 </artifactItem>
                             </artifactItems>
-                            <stripVersion>true</stripVersion>
                         </configuration>
                     </execution>
                 </executions>
             </plugin>
             <plugin>
-                <groupId>fish.payara.maven.plugins</groupId>
```
Comment (pdudits) [inline deployer/deployment-controller/pom.xml]: Notice, that with this remote setup, the server is started once for all arquillian tests, saving lots of startup time.

File: deployer/deployment-controller/pom.xml [hunk 2]
```diff
@@ -109,90 +110,49 @@
...
-                    </execution>
-                    <execution>
-                        <id>stop-test-instance</id>
-                        <phase>post-integration-test</phase>
-                        <goals>
-                            <goal>stop</goal>
-                        </goals>
-                    </execution>
-                </executions>
+                <artifactId>maven-surefire-plugin</artifactId>
```
Comment (pdudits) [inline deployer/deployment-controller/pom.xml]: Like said above, using surefire for integration tests is an antipattern (but makes sense if your entire project is made up of arquillian tests, like in samples repos or tcks).

File: deployer/deployment-controller/pom.xml [hunk 3]
```diff
@@ -109,90 +110,49 @@
...
-                    <contextRoot>/</contextRoot>
-                    <commandLineOptions>
-                        <option>
-                            <key>--nocluster</key>
-                        </option>
-                    </commandLineOptions>
+                    <configuration>
+                        <systemProperties>
+                            <skipEAR>true</skipEAR>
+                            <payara.microJar>${session.executionRootDirectory}/target/payara-micro-5.191.jar</payara.microJar>
```
Comment (pdudits) [inline deployer/deployment-controller/pom.xml]: why 191? (but it doesn't matter, we should remain using the previous setup)

File: deployer/deployment-controller/pom.xml [hunk 4]
```diff
@@ -109,90 +110,49 @@
...
+                <configuration>
+                    <rules>
+                        <requireProperty>
+                            <property>payara.version</property>
+                            <message>"Payara version must be specified."</message>
+                            <regex>^5.*</regex>
+                            <regexMessage>"This profile is only supported on Payara 5"</regexMessage>
+                        </requireProperty>
+                    </rules>
+                </configuration>
```
Comment (pdudits) [inline deployer/deployment-controller/pom.xml]: The true distribution channel for deployer will be docker image. There's no need to support multiple versions of Micro

File: deployer/deployment-controller/src/test/java/fish/payara/cloud/deployer/endpoints/UploadTest.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        return archive;
+    }
+    
+    @ArquillianResource
+    private URL base;
+    
+    //Tests that a file can be uploaded. As deployment in done asynchronously all files will deploy fine, but may be fail later 
+   @Test
+   public void uploadTest() throws IOException {
+       WebTarget jaxrstarget = javax.ws.rs.client.ClientBuilder.newClient().
```
Comment (pdudits) [inline deployer/deployment-controller/src/test/java/fish/payara/cloud/deployer/endpoints/UploadTest.java]: Is there any reason not to import ClientBuilder?

File: deployer/deployment-controller/src/test/java/fish/payara/cloud/deployer/endpoints/UploadTest.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.runner.RunWith;
+
+/**
+ * Tests that the deployment endpoint works
+ * @author jonathan coustick
+ */
+@RunWith(Arquillian.class)
+public class UploadTest {
```
Comment (pdudits) [inline deployer/deployment-controller/src/test/java/fish/payara/cloud/deployer/endpoints/UploadTest.java]: This will be then `UploadIT`

**Comment by Cousjava** [comment]
I know that the arquillian tests should be integration tests, but when I tried running them using the old setup they always failed, and I couldn't work out why after spending two days debugging it, when I switched to this setup they worked.

**Pull request merged**